internal/repository/repository.go

```diff
@@ -109,17 +112,23 @@ func (r *Repository) LoadUnpacked(ctx context.Context, buf []byte, t restic.File
 			return cerr
 		}
 		buf = wr.Bytes()
+
+		if t != restic.ConfigFile && !restic.Hash(buf).Equal(id) {
+			debug.Log("retry loading broken blob %v", h)
+			if !retriedInvalidData {
+				retriedInvalidData = true
+			} else {
+				cancel()
+			}
+			return errors.Errorf("invalid data returned")
+		}
 		return nil
 	})
 
 	if err != nil {
 		return nil, err
 	}
 
-	if t != restic.ConfigFile && !restic.Hash(buf).Equal(id) {
```

Comment:
I think removing this check is not a greatidea, because when there really is corrupt data stored in the repository, then the first time the new check above will trigger and restic will re-download the data, the second time the check won't run (due to `retriedInvalidData` being `true`), so the user will see "ciphertext verification failed", which may be scary. Thoughts?

Reply:
I think you're right. I think my idea was to also allow the caller to decide what to do if the data is damaged. But that doesn't apply here. I'll restore the check.

Reply:
Hmm, wait. The inner check in `Load` will retry once. On the second failure it will cancel the context and return with the "invalid data error". So in the worst case this returns a context canceled error. But it will return an error, such that the current line is no longer reachable if the returned data has the wrong hash.